src/lib/fetch.ts

```diff
@@ -450,83 +467,34 @@ async function fetchLastBlock(graphqlEndpoint = networkConfig.minaEndpoint) {
   return network;
 }
 
-const lastBlockQuery = `{
-  bestChain(maxLength: 1) {
-    protocolState {
-      blockchainState {
-        snarkedLedgerHash
-        stagedLedgerHash
-        date
-        utcDate
-        stagedLedgerProofEmitted
-      }
-      previousStateHash
-      consensusState {
-        blockHeight
-        slotSinceGenesis
-        slot
-        nextEpochData {
-          ledger {hash totalCurrency}
-          seed
-          startCheckpoint
-          lockCheckpoint
-          epochLength
-        }
-        stakingEpochData {
-          ledger {hash totalCurrency}
-          seed
-          startCheckpoint
-          lockCheckpoint
-          epochLength
-        }
-        epochCount
-        minWindowDensity
-        totalCurrency
-        epoch
-      }
-    }
-  }
-}`;
-
-type LastBlockQueryFailureCheckResponse = {
-  bestChain: {
-    transactions: {
-      zkappCommands: {
-        hash: string;
-        failureReason: {
-          failures: string[];
-          index: number;
-        }[];
-      }[];
-    };
-  }[];
-};
-
-const lastBlockQueryFailureCheck = (length: number) => `{
-  bestChain(maxLength: ${length}) {
-    transactions {
-      zkappCommands {
-        hash
-        failureReason {
-          failures
-          index
-        }
-      }
-    }
+async function fetchCurrentSlot(graphqlEndpoint = networkConfig.minaEndpoint) {
```

Comment:
This was implemented but isn't used (I was planning to implement `getCurrentSlot`, but it will wait for async circuits), and will most likely revert.

Reply:
seems useful to me! with async circuits we can just use this new function directly from a method